godot/src/logic/content_thread.gd

```diff
@@ -0,0 +1,504 @@
+extends RefCounted
+class_name ContentThread
+
+enum ContentType {
+	CT_GLTF_GLB = 1,
+	CT_TEXTURE = 2,
+	CT_WEARABLE_EMOTE = 3,
+	CT_MESHES_MATERIAL = 4,
+	CT_INSTACE_GLTF = 5,
+	CT_AUDIO = 6,
+	CT_VIDEO = 7
+}
+
+var pending_content: Array[Dictionary] = []
+var http_requester = RustHttpRequesterWrapper.new()
+var thread: Thread = null  # null=main_thread
+var id: int = -1
+
+
+func append_content(request: Dictionary):
+	pending_content.push_back(request)
+
+
+func content_processing_count():
+	return pending_content.size()
+
+
+func _init(id: int, thread: Thread):
+	self.thread = thread
+	self.id = id
+
+
+func process(content_cache_map: Dictionary, wearable_cache_map: Dictionary):
+	http_requester.poll()
```

Comment:
I don't understand this, if it supposes to use co-routines below why it needs to call to poll after the entire resource fetching?

Reply:
You are right.
The `process` is not being awaited correctly. I changed the code of the process to:

```gdscript
func process(content_cache_map: Dictionary):  # not a coroutine
	_http_requester.poll()

	if _pending_content.is_empty():
		return

	var loading_content: Array[Dictionary] = []
	while _pending_content.size() > 0:
		loading_content.push_back(_pending_content.pop_front())

	for content in loading_content:
		# coroutine not awaited, it can process multiple times
		co_process_content(content, content_cache_map)
```

So now, it's spawns `co_process_content` as many times as needed in the thread, and they are processed using the `_http_requester`.